### PR TITLE
LPD-34191 Upgrade DDMField and DDMFieldAttribut company id when is 0

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
LPD-34191 After JournalArticleDDMFieldsUpgradeProcess DDMField and DDMFieldAttributes has a 0 on the field companyId

# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-34191

`On the com.liferay.journal.internal.upgrade.v4_0_0.JournalArticleDDMFieldsUpgradeProcess` if needed, the company id on the creation of the `ddmField` and `ddmFieldAttribute` is obtained from the `CompanyThreadLocal.getCompanyId()` but because is a upgrade this value is 0.

# How am I fixing it?

On the `com.liferay.journal.internal.upgrade.v4_0_0.JournalArticleDDMFieldsUpgradeProcess` we should fill with the company data from the database.
And because the database can have data with values filled with 0 on `ddmField` and `ddmFieldAttribute`, 

We create upgrades for each table with a process to fill this values with the values from the ddmStructure.

# How can you verify that it works?

Run the included automated tests.

# Commits

- **LPD-34191 create upgrade to fill the incorrect data that can be generated by error**
- **LPD-34191 register the new upgrades**
- **LPD-34191 add test to check the new upgrades**

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:report:63547464

**Comment by AliciaGarciaGarcia** [comment]
ci:report:63534495

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:report:65367943

**Comment by AliciaGarciaGarcia** [comment]
ci:report:65437260

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldAttributeUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+				connection.prepareStatement(
+					SQLTransformer.transform(
+						StringBundler.concat(
+							"select DDMFieldAttribute.ctCollectionId, ",
+							"DDMFieldAttribute.fieldAttributeId, ",
+							"DDMFieldAttribute.fieldId, ",
+							"DDMField.structureVersionId from ",
+							"DDMFieldAttribute inner join DDMField on ",
+							"DDMFieldAttribute.fieldId = DDMField.fieldId ",
+							"where DDMFieldAttribute.companyId = 0")));
```
Comment (adolfopa) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldAttributeUpgradeProcess.java:43]: Just making sure: shouldn't you add a `or is null` in the `where` condition as well? If the company ID column is unset you may want to update it as well.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldAttributeUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+			ResultSet resultSet = selectPreparedStatement.executeQuery();
+
+			while (resultSet.next()) {
+				long structureVersionId = resultSet.getLong(
+					"structureVersionId");
+
+				DDMStructureVersion ddmStructureVersion =
+					_ddmStructureVersionLocalService.fetchDDMStructureVersion(
+						structureVersionId);
```
Comment (adolfopa) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldAttributeUpgradeProcess.java:58]: You cannot use the services here. These services "belong" to the tables you're upgrading. You should use an SQL query here.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldAttributeUpgradeProcess.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+					if (_log.isWarnEnabled()) {
+						_log.warn(
+							StringBundler.concat(
+								"Change the company id value for field ",
+								"Attribute ID ",
+								resultSet.getLong("fieldAttributeId"),
+								" from 0 to ",
+								ddmStructureVersion.getCompanyId()));
+					}
```
Comment (adolfopa) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldAttributeUpgradeProcess.java:86]: Why is this at WARN level? Shouldn't this be at DEBUG or INFO level?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldUpgradeProcess.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+			ResultSet resultSet = selectPreparedStatement.executeQuery();
+
+			while (resultSet.next()) {
+				long structureVersionId = resultSet.getLong(
+					"structureVersionId");
+
+				DDMStructureVersion ddmStructureVersion =
+					_ddmStructureVersionLocalService.fetchDDMStructureVersion(
+						structureVersionId);
```
Comment (adolfopa) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldUpgradeProcess.java:52]: As before, you shouldn't use the DDM services in this upgrade.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/test/DDMFieldAttributeUpgradeProcessTest.java [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+			ddmFieldAttributes = _ddmFieldLocalService.getDDMFieldAttributes(
+				_STORAGE_ID, ddmFieldAttributeName);
+
+			for (DDMFieldAttribute ddmFieldAttribute : ddmFieldAttributes) {
+				Assert.assertNotEquals(0L, ddmFieldAttribute.getCompanyId());
+			}
+		}
+	}
+
+	private void _addDDMValues() throws Exception {
```
Comment (adolfopa) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/test/DDMFieldAttributeUpgradeProcessTest.java:121]: Can you rename this to `_addDDMFormValues`?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldAttributeUpgradeProcess.java (hunk 2 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldAttributeUpgradeProcess.java:58]: thanks!

I was working on it :D

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldUpgradeProcess.java (hunk 4 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldUpgradeProcess.java:52]: I do not know WHY but in my head ddmStructure* was an external service from ddm.... XD

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/test/DDMFieldAttributeUpgradeProcessTest.java (hunk 5 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/test/DDMFieldAttributeUpgradeProcessTest.java:121]: thanks!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldAttributeUpgradeProcess.java (hunk 3 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldAttributeUpgradeProcess.java:86]: thanks!

**Comment by AliciaGarciaGarcia** [comment]
ci:test:stable

**Review by adolfopa — APPROVED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldAttributeUpgradeProcess.java (hunk 1 above)
Comment (adolfopa) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldAttributeUpgradeProcess.java]: The companyId column has a not-null restriction, so this is not needed.

**Comment by adolfopa** [comment]
@jkappler Not forwarding, as you have this assigned.

**Comment by AliciaGarciaGarcia** [comment]
ci:test:upgrade

**Comment by AliciaGarciaGarcia** [comment]
ci:reevaluate:1267401_13800

**Comment by AliciaGarciaGarcia** [comment]
ci:report:66583589

**Comment by AliciaGarciaGarcia** [comment]
the only test that make me wonder is : 
[LocalFile.DatabasePartitioning#CanSetVirtualHostViaPages](https://testray.liferay.com/#/project/34754/routines/34756/build/66583589/case-result/66603252/history)

relaunching upgrade test

**Comment by AliciaGarciaGarcia** [comment]
ci:test:upgrade

**Comment by AliciaGarciaGarcia** [comment]
ci.test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:report:67572202

**Comment by AliciaGarciaGarcia** [comment]
failing test PortalLogAssertorTest-modules-integration-mysql57-jdk8/0/8 seems relevant : https://testray.liferay.com/#/project/34754/routines/34756/build/67572202/case-result/67576606/history

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
Hi @adolfopa 
https://github.com/liferay-content-management/liferay-portal/pull/5765#issuecomment-2301773702 was a strange error that I do not know if with my 2 last commits I will solve.

if relevant does not fail, can you re-review this PR (the las 2 commits only) to see If my changes are not wrong?

thanks!!

**Comment by AliciaGarciaGarcia** [comment]
ci:test:upgrade

**Comment by AliciaGarciaGarcia** [comment]
upgrade failures seem not related with this PR.

@adolfopa If you can review the last 2 commits and is ok, we can forward this.

Thanks!

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**